mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/ListeningCurrentQuotaUpdater.java

```diff
@@ -87,7 +90,7 @@ private void handleExpungedEvent(Expunged expunged, QuotaRoot quotaRoot) throws
         long expungedCount = (long) expunged.getUids().size();
         // Expunge event can contain no data (expunge performed while no messages marked \Deleted)
         if (expungedCount != 0 && expungedSize != 0) {
-            currentQuotaManager.decrease(quotaRoot, expungedCount, expungedSize);
+            currentQuotaManager.decrease(new CurrentQuota(quotaRoot, QuotaCountUsage.count(expungedCount), QuotaSizeUsage.size(expungedSize)));
```

Comment:
Should I better create the `CurrentQuota` above and write a function in the pojo returning if it's an empty quota (both count and usage at 0) or not?

Reply:
IMO the `CurrentQuotaDiff` object should only accept strictly positive numbers.

In this method we should have a 

`Optional<CurrentQuotaDiff> computeCurrentQuotaDiff(Expunged expunged)`

And here we would write simply

```
omputeCurrentQuotaDiff(expunged).ifPresent(currentQuotaManager::decrease);
```

Reply:
it could. But just look for a minute: what you pass to `decrease` is not the CurrentQuota of a QuotaRoot, it's a QuotaOperation or something like that (it aims at modifying the CurrentQuota of a QuotaRoot but is not a Current Quota. Am I right?

Reply:
I like `QuotaOperation`, thanks for the suggestion !